runners/spark/src/main/java/org/apache/beam/runners/spark/translation/streaming/CheckpointDir.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.runners.spark.translation.streaming;
+
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Spark checkpoint dir tree.
```

Comment:
Perhaps it's worth noting here that the runner defines it's own "parent" checkpoint-dir, containing Spark's checkpoint-dir and other things it checkpoints on it's own (e.g., Accumulators for fault-tolerance).

Also worth noting that in the context of Spark's (and it's runner) checkpointing all underlying FSs use Hadoop's `FileSystem` as a Spark perquisite with GS/S3 supporting via specific connectors. Adding checkpoints to the runner keeps the same behaviour (not relying on Beam's FS API for example) to be consistent with Spark.